src/genie_python/block_names.py

```diff
@@ -45,44 +53,49 @@ def update_prefix(self, pv_prefix):
                 self._pv_name = new_name
                 if self._timer is None:
                     self._timer = Timer(0, self._add_monitor)
-                    self._timer.setDaemon(True)
+                    self._timer.daemon = True
                     self._timer.start()
 
-    def _add_monitor(self):
+    def _add_monitor(self) -> None:
         """
-        Add a monitor to the block names pv if it is not already monitored, then reschedule task to add monitor. If the
-        pv is monitored don't run.
+        Add a monitor to the block names pv if it is not already monitored,
+        then reschedule task to add monitor. If the pv is monitored don't run.
         """
 
-        # Get PV we should monitor and check whether we need to add new monitor if so cancel old monitor
+        # Get PV we should monitor and check whether we need to add new monitor
+        # if so cancel old monitor
         with self.pv_name_lock:
             self._timer = None
             # not monitoring the correct pv
             if self._cancel_monitor_fn is not None:
                 self._cancel_monitor_fn()
 
-            # Add new monitor if successful then record monitored pv and pull first value, otherwise do nothing
+            # Add new monitor if successful then record monitored pv and pull
+            # first value, otherwise do nothing
             try:
                 self._cancel_monitor_fn = CaChannelWrapper.add_monitor(
                     self._pv_name, self._update_block_names, to_string=True
                 )
                 self._update_block_names(
-                    CaChannelWrapper.get_pv_value(self._pv_name, to_string=True), None, None
+                    CaChannelWrapper.get_pv_value(self._pv_name, to_string=True), "", ""
                 )
             except UnableToConnectToPVException:
-                # Schedule next add monitor if needed; i.e. a old pv-prefix change was slower the the last pv-prefix
+                # Schedule next add monitor if needed; i.e. a old pv-prefix change
+                # was slower the the last pv-prefix
                 # and so we are monitoring the wrong pv.
                 self._timer = Timer(self._delay_before_retry_add_monitor, self._add_monitor)
-                self._timer.setDaemon(True)
+                self._timer.daemon = True
                 self._timer.start()
 
-    def _update_block_names(self, value, _, _1):
+    def _update_block_names(self, value: PVValue[E], _: str, _1: str) -> None:
```

Comment:
This one is quite mean of me, but I have just updated the add_monitor of ca_channel and genie_pv_connection_protocol to using Optional[str] for compatibility reasons, we'll need this to be optional for _ and _1 here for the same reasons. This will make your pyright fail, but only until my change is reviewed.

Reply:
https://github.com/ISISComputingGroup/genie/pull/36/files for reference